torch/csrc/jit/mobile/interpreter.cpp

```diff
@@ -191,8 +191,10 @@ bool InterpreterState::run(Stack& stack) {
         ++pc;
       } break;
       case NAMED_TUPLE_CONSTRUCT: {
-        auto type = code_->types_[inst.X]->expect<at::TupleType>();
-        namedTupleConstruct(stack, type, inst.N);
+        namedTupleConstruct(
```

Comment:
Should we also do this in DICT_CONSTRUCT and LIST_CONSTRUCT?

Reply:
No, `listConstruct` takes const reference to `ListType` (i.e., not a shared_ptr) and `dictConstruct` takes const reference to `DictTypePtr`. It does seem weird to have `namedTupleConstruct` and `dictConstruct` both taking shared_ptrs, but not agree on whether it's by value or by reference. After looking, probably `dictConstruct` should take const reference to`DictType` too.